### ISSUE TITLE
Sidecar registers wrong port to eureka?

### ISSUE THREAD

**Issue by edevrieze** [spring-cloud/spring-cloud-netflix#225]
Hi, 

I'm trying to a get a JVM-basd service to talk to a non-JVM-based service which uses a sidecar process to enable registration with eureka.

However, I'm noticed that the port registered with eureka for the non-JVM-based service is the one of the sidecar process itself, instead of the port on which the non-JVM-based service is listening.

So basically, looking at my application.yml (below), I would expect the sidecar process to have registered port 8000 with eureka, rather than 8001 (which is the sidecar port itself). As a result, calling the non-jvm service using the ribbon-enabled ResTemplate fails, because it is trying to send the request to port 8001, rather than 8000.

Am I missing something?

Thanks,
Erik

application.yml:
```
server:
  port: 8001
spring:
  application:
    name: MANUFACTURERS

sidecar:
  port: 8000
  health-uri: http://localhost:8000/health
```

Eureka (http://localhost:8761/eureka/apps)
```xml
<application>
  <name>MANUFACTURERS</name>
  <instance>
    <hostName>LVNLT9QZHJ12</hostName>
    <app>MANUFACTURERS</app>
    <ipAddr>10.100.222.21</ipAddr>
    <status>UP</status>
    <overriddenstatus>UNKNOWN</overriddenstatus>
    <port enabled="true">8001</port>
    <securePort enabled="false">443</securePort>
    <countryId>1</countryId>
    <dataCenterInfo class="com.netflix.appinfo.InstanceInfo$DefaultDataCenterInfo">
      <name>MyOwn</name>
    </dataCenterInfo>
    <leaseInfo>
      <renewalIntervalInSecs>30</renewalIntervalInSecs>
      <durationInSecs>90</durationInSecs>
      <registrationTimestamp>1424689311672</registrationTimestamp>
      <lastRenewalTimestamp>1424696396125</lastRenewalTimestamp>
      <evictionTimestamp>0</evictionTimestamp>
      <serviceUpTimestamp>1424689281608</serviceUpTimestamp>
    </leaseInfo>
    <metadata class="java.util.Collections$EmptyMap"/>
    <appGroupName>MYSIDECARGROUP</appGroupName>
    <homePageUrl>http://LVNLT9QZHJ12:8000/</homePageUrl>
    <statusPageUrl>http://LVNLT9QZHJ12:8001/info</statusPageUrl>
    <healthCheckUrl>http://LVNLT9QZHJ12:8001/health</healthCheckUrl>
    <vipAddress>MANUFACTURERS</vipAddress>
    <isCoordinatingDiscoveryServer>true</isCoordinatingDiscoveryServer>
    <lastUpdatedTimestamp>1424689311672</lastUpdatedTimestamp>
    <lastDirtyTimestamp>1424689291504</lastDirtyTimestamp>
    <actionType>ADDED</actionType>
  </instance>
</application>
```

**Comment by dsyer** [comment]
Could be an ordering problem with the config files.

**Comment by tgolden-andplus** [comment]
It's likely that I'm misunderstanding how the Sidecar is supposed to work or be configured, but I'm running into this same issue.

My sidecar app declares itself as a `@SpringBootApplication` with `@EnableSidecar`. I've disabled the remote config server and am just configuring everything in `application.properties`, which looks like:

```
spring.application.name=todo-service
server.port=8084

sidecar.port=8083
sidecar.health-uri=http://localhost:${sidecar.port}/health
```

When I start up the sidecar app and refresh Eureka, I see that it's been registered with the 8084 port. My understanding is it ought to be using 8083, based on my configuration - otherwise, how could I set up Zuul to point to the todo-service? I'd want to configure Zuul like:

```
zuul.routes.todos.path=/todos/**
zuul.routes.todos.serviceId=todo-service
```

but it appears to be trying to talk to the sidecar app directly on 8084 instead of the app it's supposed to proxy on 8083.

This is all using Camden.SR3.

**Comment by yankhonskikendaxa** [comment]
I solved this issue by having the same sidecar port as the application port. So the combination of Sidecar + app is discovered as a single application (with sidecar port) in the registry (Eureka server).

Then I configured zuul settings in the sidecar, so it redirects all request with given pattern to the application.

Then I excluded some technical path from zuul, so Spring Cloud end-points (Spring Boot actuator)
